src/ucp/rma/basic_rma.c

```diff
@@ -25,18 +25,81 @@
         return UCS_ERR_INVALID_PARAM; \
     }
 
+#define _USE_ZCOPY 1
+
 ucs_status_t ucp_put(ucp_ep_h ep, const void *buffer, size_t length,
                      uint64_t remote_addr, ucp_rkey_h rkey)
 {
     ucp_ep_rma_config_t *rma_config;
     ucs_status_t status;
     uct_rkey_t uct_rkey;
-    size_t frag_length;
     ssize_t packed_len;
     ucp_lane_index_t lane;
 
     UCP_RMA_CHECK_PARAMS(buffer, length);
 
+#if _USE_ZCOPY
+    ucp_ep_config_t *config;
+
+    UCP_EP_RESOLVE_RKEY(ep, rkey, rma, config, lane, uct_rkey);
+    rma_config = &config->rma[lane];
+
+    if (length <= rma_config->max_put_short) {
+        status = uct_ep_put_short(ep->uct_eps[lane], buffer,
+                                  length, remote_addr,
+                                  uct_rkey);
+        ucp_worker_progress(ep->worker);
+    } else  if (length <= rma_config->max_put_bcopy) {
+        ucp_memcpy_pack_context_t pack_ctx;
+        pack_ctx.src    = buffer;
+        pack_ctx.length = length;
+        packed_len = uct_ep_put_bcopy(ep->uct_eps[lane],
+                                      ucp_memcpy_pack, &pack_ctx,
+                                      remote_addr, uct_rkey);
+
+        status = (packed_len > 0) ? UCS_OK : (ucs_status_t)packed_len;
+        ucp_worker_progress(ep->worker);
+    } else {
+        uct_completion_t uct_comp;
+        uct_mem_h mem = NULL;
+        uct_md_h uct_md = ucp_ep_md(ep, lane);
+
+        uct_comp.count = 2;
```

Comment:
need to consider case where length is larger than max_zcopy supported by that UCT